[FEAT]: implement KvdbApi

Doominika opened the pull request:
## Description
Implement KvdbApi and data models used by it.

## PrivMX Endpoint

### ADDITIONS

**New classes:**
- KVDB (new module)
  - `KvdbApi`
  - `Kvdb`
  - `ServerKvdbEntryInfo`
  - `KvdbEntry`
  - `KvdbDeletedEventData`
  - `KvdbStatsEventData`
  - `KvdbDeletedEntryEventData`

## PrivMX Endpoint

### ADDITIONS
Added support for new `KvdbApi` module.

Doominika committed:
feat: implement KvdbApi, types and events

Doominika committed:
refactor: change names of classes, methods and elements using kvdbEntry instead Item

Doominika committed:
refactor: update Kvdb class

Doominika committed:
refactor: update ServerKvdbEntryInfo class

Doominika committed:
refactor: update KvdbEntry class

Doominika committed:
refactor: update KvdbKeysPagingQuery class

Doominika committed:
refactor: rename ItemsPagingQuery class to KvdbEntryPagingQuery and update related files

Doominika committed:
docs: fix typo in KvdbEntry constructor description

Doominika committed:
feat: add kvdb to CMakeLists

Doominika committed:
refactor: add missing return in jni listKvdbs

Doominika committed:
refactor: update KvdbDeletedItemEventData class

Doominika committed:
refactor: update KvdbStatsEventData class

Doominika committed:
refactor: fix function name mismatch between header and implementation

Doominika committed:
docs: add description to KvdbApi

Doominika committed:
feat: add EventTypes for kvdb

Doominika committed:
feat(privmx-endpoint-extra): add new module KvdbApi

Doominika committed:
docs: add description to KvdbApi methods

Doominika committed:
refactor: delete KvdbKeysPagingQuery and KvdbEntryPagingQuery and update KvdApi to 2.5

Doominika committed:
refactor: change pagingQuery argunent in listKvdbs to parameters

Doominika committed:
chore: format code for KvdbApi

Doominika committed:
chore: remove PagingQuery

Doominika committed:
fix: add missing arguments to kvdb2Java and kvdbEntry2Java in jni

Doominika committed:
chore: remove parsers KvdbKeysPagingQuery and KvdbEntryPagingQuery

Doominika committed:
refactor: make deleteEntries mehod in KvdbApi class return map

Doominika committed:
chore: delete unused declarations in parser.h file

Doominika committed:
docs: add missing descriptions to createKvdb method

Doominika committed:
refactor: make Kvdb parameters public and change long to Long

Doominika committed:
refactor: make KvdbEntry parameters public and change long to Long

Doominika committed:
refactor: make ServerKvdbEntryInfo parameters public and change long to Long

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/model/events/KvdbDeletedEntryEventData.java:
```diff
@@ -0,0 +1,22 @@
+    /**
+     * Kvdb ID
+     */
+    String kvdbId;
```
add modifier "public" to fields

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/model/events/KvdbDeletedEventData.java:
```diff
@@ -0,0 +1,16 @@
+    /**
+     * Kvdb ID
+     */
+    String kvdbId;
```
add "public" modifier

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/model/events/KvdbStatsEventData.java:
```diff
@@ -0,0 +1,28 @@
+    /**
+     * Kvdb ID
+     */
+    String kvdbId;
```
add "public" modifier to fields

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/model/Kvdb.java:
```diff
@@ -0,0 +1,120 @@
+    /**
+     * ID of user who created the Kvdb
+     */
+    String creator;
```
add public modifier

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/model/Kvdb.java:
```diff
@@ -0,0 +1,120 @@
+    /**
+     * Kvdb last modification timestamp
+     */
+    public Long lastModificationDate;
```
sometimes you use Long but sometimes long type in models

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/model/Kvdb.java:
```diff
@@ -0,0 +1,120 @@
+     * @param statusCode           Retrieval and decryption status code
+     * @param schemaVersion        Version of the Kvdb data structure and how it is encoded/encrypted
+     */
+    public Kvdb(String contextId, String kvdbId, Long createDate, String creator, Long lastModificationDate, String lastModifier, List<String> users, List<String> managers, Long version, byte[] publicMeta, byte[] privateMeta, Long entries, Long lastEntryDate, ContainerPolicy policy, Long statusCode, Long schemaVersion) {
```
reformat this long list of parameters

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/model/events/KvdbDeletedEntryEventData.java:
```diff
@@ -0,0 +1,22 @@
+
+/**
+ * Holds information of `KvdbDeletedEntryEvent`.
+ */
```
Event data models has these additional tags
```
* @category core
* @group Events
```

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/model/events/KvdbDeletedEventData.java:
```diff
@@ -0,0 +1,16 @@
+
+/**
+ * Holds information of `KvdbDeletedEvent`.
+ */
```
Event data models has these additional tags
```
* @category core
* @group Events
```

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/model/events/KvdbStatsEventData.java:
```diff
@@ -0,0 +1,28 @@
+
+/**
+ * Holds Kvdb statistical data.
+ */
```
Event data models has these additional tags
```
* @category core
* @group Events
```

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/model/Kvdb.java:
```diff
@@ -0,0 +1,120 @@
+/**
+ * Holds all available information about a Kvdb.
+ *
+ * @category kvdb
```
this model should be grouped in kvdb group

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/model/KvdbEntry.java:
```diff
@@ -0,0 +1,71 @@
+ * Holds all available information about a Entry.
+ *
+ * @category kvdb
+ */
```
this model should be grouped in kvdb group

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/model/ServerKvdbEntryInfo.java:
```diff
@@ -0,0 +1,43 @@
+ * Holds Kvdb entry's information created by the server.
+ *
+ * @category kvdb
+ */
```
this model should be grouped in kvdb group

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/model/KvdbEntry.java:
```diff
@@ -0,0 +1,71 @@
+     * @param statusCode    Retrieval and decryption status code
+     * @param schemaVersion Version of the Entry data structure and how it is encoded/encrypted
+     */
+    public KvdbEntry(ServerKvdbEntryInfo info, byte[] publicMeta, byte[] privateMeta, byte[] data, String authorPubKey, Long version, Long statusCode, Long schemaVersion) {
```
reformat list of parameters

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * @throws PrivmxException       thrown when method encounters an exception.
+     * @throws NativeException       thrown when method encounters an unknown exception.
+     */
+    public native void unsubscribeFromKvdbEvents(String kvdbId) throws PrivmxException, NativeException, IllegalStateException;
```
```suggestion
    public native void unsubscribeFromEntryEvents(String kvdbId) throws PrivmxException, NativeException, IllegalStateException;
```

djenczewski commented on privmx-endpoint-extra/src/main/java/com/simplito/java/privmx_endpoint_extra/lib/PrivmxEndpoint.java:
```diff
@@ -194,6 +206,18 @@ private void unsubscribeChannel(String channelStr) {
+        if (channel.module.startsWith("kvdb") && kvdbApi != null) {
+            if (channel.type != null && channel.type.equals("entries")) {
+                if (channel.instanceId != null) {
+                    kvdbApi.unsubscribeFromKvdbEvents(channel.instanceId);
```
```suggestion
kvdbApi.unsubscribeFromEntryEvents(channel.instanceId);
```

djenczewski commented on privmx-endpoint/src/main/cpp/modules/KvdbApi.h:
```diff
@@ -0,0 +1,14 @@
+//
```
unnecessary file

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+
+/**
+ * Manages PrivMX Bridge  Kvdbs and their messages.
+ */
```
add category kvdb

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * Creates an instance of {@code KvdbApi}.
+     *
+     * @param connection instance of 'Connection'
+     * @throws IllegalStateException thrown when instance is closed.
```
how instance could be closed during initialization?

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * Creates a new KVDB in given Context.
+     *
+     * @param contextId   ID of the Context to create the KVDB in
+     * @param users       array of UserWithPubKey structs which indicates who will have access to the created KVDB
```
```suggestion
     * @param users       list of {@link UserWithPubKey} which indicates who will have access to the created KVDB
```

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     *
+     * @param contextId   ID of the Context to create the KVDB in
+     * @param users       array of UserWithPubKey structs which indicates who will have access to the created KVDB
+     * @param managers    array of UserWithPubKey structs which indicates who will have access (and management rights) to the created KVDB
```
```suggestion
     * @param managers    list of {@link UserWithPubKey} which indicates who will have access (and management rights) to the created KVDB
```

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * Creates a new KVDB in given Context.
+     *
+     * @param contextId   ID of the Context to create the KVDB in
+     * @param users       array of UserWithPubKey structs which indicates who will have access to the created KVDB
```
```suggestion
     * @param users       list of {@link UserWithPubKey} which indicates who will have access to the created KVDB
```

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     *
+     * @param contextId   ID of the Context to create the KVDB in
+     * @param users       array of UserWithPubKey structs which indicates who will have access to the created KVDB
+     * @param managers    array of UserWithPubKey structs which indicates who will have access (and management rights) to the created KVDB
```
```suggestion
     * @param managers    list of {@link UserWithPubKey} which indicates who will have access (and management rights) to the created KVDB
```

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     *
+     * @param kvdbId              ID of the KVDB to update
+     * @param users               array of UserWithPubKey structs which indicates who will have access to the created KVDB
+     * @param managers            array of UserWithPubKey structs which indicates who will have access (and management rights) to the created KVDB
```
```suggestion
     * @param managers    list of {@link UserWithPubKey} which indicates who will have access (and management rights) to the created KVDB
```

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     *
+     * @param kvdbId              ID of the KVDB to update
+     * @param users               array of UserWithPubKey structs which indicates who will have access to the created KVDB
+     * @param managers            array of UserWithPubKey structs which indicates who will have access (and management rights) to the created KVDB
```
```suggestion
     * @param managers    list of {@link UserWithPubKey} which indicates who will have access (and management rights) to the created KVDB
```

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * Updates an existing KVDB.
+     *
+     * @param kvdbId              ID of the KVDB to update
+     * @param users               array of UserWithPubKey structs which indicates who will have access to the created KVDB
```
```suggestion
     * @param users       list of {@link UserWithPubKey} which indicates who will have access to the created KVDB
```

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * Updates an existing KVDB.
+     *
+     * @param kvdbId              ID of the KVDB to update
+     * @param users               array of UserWithPubKey structs which indicates who will have access to the created KVDB
```
```suggestion
     * @param users       list of {@link UserWithPubKey} which indicates who will have access to the created KVDB
```

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * @throws PrivmxException       thrown when method encounters an exception.
+     * @throws NativeException       thrown when method encounters an unknown exception.
+     */
+    public void updateKvdb(
```
create overloads for force and forceGenerateNewKey params

Doominika replied:
Question ...

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * Gets a KVDB by given KVDB ID.
+     *
+     * @param kvdbId ID of KVDB to get
+     * @return struct containing info about the KVDB
```
we haven't structs

djenczewski replied:
this method don't return class but object

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * @param skip        skip number of elements to skip from result
+     * @param limit       limit of elements to return for query
+     * @param sortOrder   order of elements in result ("asc" for ascending, "desc" for descending)
+     * @param lastId      ID of the element from which query results should start\
```
trailing "\"?

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * @param skip        skip number of elements to skip from result
+     * @param limit       limit of elements to return for query
+     * @param sortOrder   order of elements in result ("asc" for ascending, "desc" for descending)
+     * @param lastId      ID of the element from which query results should start\
```
trailing "\"?

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * @param skip      skip number of elements to skip from result
+     * @param limit     limit of elements to return for query
+     * @param sortOrder order of elements in result ("asc" for ascending, "desc" for descending)
+     * @param lastId    ID of the element from which query results should start\
```
trailing "\"?

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * @param lastId      ID of the element from which query results should start\
+     * @param queryAsJson stringified JSON object with a custom field to filter result
+     * @param sortBy      field by elements are sorted in result
+     * @return struct containing a list of Kvdbs
```
we haven't structs

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * @param sortOrder   order of elements in result ("asc" for ascending, "desc" for descending)
+     * @param lastId      ID of the element from which query results should start\
+     * @param queryAsJson stringified JSON object with a custom field to filter result
+     * @return struct containing a list of Kvdbs
```
we haven't structs

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * @param limit     limit of elements to return for query
+     * @param sortOrder order of elements in result ("asc" for ascending, "desc" for descending)
+     * @param lastId    ID of the element from which query results should start\
+     * @return struct containing a list of Kvdbs
```
we haven't structs

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * @param skip      skip number of elements to skip from result
+     * @param limit     limit of elements to return for query
+     * @param sortOrder order of elements in result ("asc" for ascending, "desc" for descending)
+     * @return struct containing a list of Kvdbs
```
we haven't structs

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * @param skip        skip number of elements to skip from result
+     * @param limit       limit of elements to return for query
+     * @param sortOrder   order of elements in result ("asc" for ascending, "desc" for descending)
+     * @param lastId      ID of the element from which query results should start\
```
trailing "\"?

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,553 @@
+     * @throws PrivmxException       thrown when method encounters an exception.
+     * @throws NativeException       thrown when method encounters an unknown exception.
+     */
+    public native HashMap<String, Boolean> deleteEntries(
```
```suggestion
    public native Map<String, Boolean> deleteEntries(
```

djenczewski commented on privmx-endpoint/src/main/cpp/modules/KvdbApi.cpp:
```diff
@@ -0,0 +1,616 @@
+        jstring sort_by
+) {
+    JniContextUtils ctx(env);
+    if (ctx.nullCheck(context_id, "Context ID")) {
```
add null check for sortOrder

djenczewski commented on privmx-endpoint/src/main/cpp/modules/KvdbApi.cpp:
```diff
@@ -0,0 +1,616 @@
+        jstring sort_by
+) {
+    JniContextUtils ctx(env);
+    if (ctx.nullCheck(kvdb_id, "Kvdb ID")) {
```
add nullcheck for sort_order

djenczewski commented on privmx-endpoint/src/main/cpp/modules/KvdbApi.cpp:
```diff
@@ -0,0 +1,616 @@
+                    query.sortBy = ctx.jString2string(sort_by);
+                }
+
+                auto itemKeys_c(
```
replace itemKeys with entryKeys

djenczewski commented on privmx-endpoint/src/main/cpp/modules/KvdbApi.cpp:
```diff
@@ -0,0 +1,616 @@
+                    query.sortBy = ctx.jString2string(sort_by);
+                }
+
+                auto items_c(
```
replace items with entries

djenczewski commented on privmx-endpoint/src/main/cpp/modules/KvdbApi.cpp:
```diff
@@ -0,0 +1,616 @@
+        jstring key
+) {
+    JniContextUtils ctx(env);
+    if (ctx.nullCheck(kvdb_id, "Kvdb ID")) {
```
add nullcheck for key

djenczewski commented on privmx-endpoint/src/main/cpp/modules/KvdbApi.cpp:
```diff
@@ -0,0 +1,616 @@
+    if (ctx.nullCheck(kvdb_id, "Kvdb ID") ||
+        ctx.nullCheck(key, "Key") ||
+        ctx.nullCheck(public_meta, "Public meta") ||
+        ctx.nullCheck(private_meta, "Private meta")) {
```
add nullcheck for data

djenczewski commented on privmx-endpoint/src/main/cpp/modules/KvdbApi.cpp:
```diff
@@ -0,0 +1,616 @@
+
+                for (int i = 0; i < ctx->GetArrayLength(keys_arr); i++) {
+                    jobject arrayElement = ctx->GetObjectArrayElement(keys_arr, i);
+                    keys_c.push_back(ctx.jString2string((jstring) arrayElement));
```
add check for null arrayElement

djenczewski commented on privmx-endpoint/src/main/cpp/parser.cpp:
```diff
@@ -422,4 +422,34 @@ parseEvent(JniContextUtils &ctx, std::shared_ptr<privmx::endpoint::core::Event>
+    auto limit = ctx.getObject(ctx->GetObjectField(pagingQuery, limitFID)).getLongValue();
+    auto sortOrder = ctx.jString2string((jstring) ctx->GetObjectField(pagingQuery, sortOrderFID));
+
+    result.skip = skip;
```
skip, limit and sortOrder can be directly assigned to result fields

djenczewski commented on privmx-endpoint/src/main/cpp/parser.cpp:
```diff
@@ -422,4 +422,34 @@ parseEvent(JniContextUtils &ctx, std::shared_ptr<privmx::endpoint::core::Event>
+}
+
+privmx::endpoint::core::PagingQuery
+parsePagingQuery(JniContextUtils &ctx, jobject pagingQuery) {
```
this method should be extended by sortBy field

djenczewski reviewed (changes requested):
1. Remove trailing "\" i lastID param description in KvdbApi methods
2. replace struct word with object
3. reformat cpp code
4. Add nullchecks for non optional parameters with reference type

Doominika committed:
docs: add tags to KvdbDeletedEntryEventData class

Doominika committed:
docs: add tags to KvdbDeletedEventData class

Doominika committed:
docs: add tags to KvdbStatsEventData class

Doominika committed:
docs: add public modifier to KvdbDeletedEntryEventData class fields

Doominika committed:
docs: add public modifier to KvdbDeletedEventData class fields

Doominika committed:
docs: add public modifier to KvdbStatsEventData class fields

Doominika committed:
refactor: add public modifier to Kvdb class field

Doominika committed:
refactor: change long to Long in KvdbStatsEventData class

Doominika committed:
chore: reformat Kvdb constructor

Doominika committed:
chore: reformat KvdbEntry constructor

Doominika committed:
docs: add group tag to Kvdb class

Doominika committed:
docs: add group tag to KvdbEntry class

Doominika committed:
docs: add group tag to ServerKvdbEntryInfo class

Doominika committed:
fix: fix methods names in KvdbApi

Doominika committed:
fix(privmx-enpoint-extra): use correct method from KvdbApi

Doominika committed:
chore: delete KvdbApi.h file

Doominika committed:
docs: add category tag to KvdbApi class

Doominika committed:
docs: correct description of KvdbApi constructor thrown exception

Doominika committed:
docs: change word array to list and remove structs references

Doominika committed:
feat: create overloads for force and forceGenerateNewKey params

Doominika committed:
docs: remove structs references in descriptions

Doominika committed:
chore: change HashMap to Map

Doominika committed:
feat: add null check for sortOrder in KvdbApi in jni

Doominika committed:
chore: change variable names from items to entries

Doominika committed:
feat: add null checking in KvdbApi class in jni

Doominika committed:
feat: add new sortBy parameter to parsePagingQuery method

Doominika committed:
docs: remove unnecessary slash characters

Doominika committed:
docs: replace word class with word object

Doominika committed:
chore: reformat cpp code

Doominika committed:
chore: remove default force parameter value from updateKvdb method

Doominika committed:
docs: remove unnecessary slash characters

Doominika committed:
feat: add new hasEntry method to KvdbApi

Doominika committed:
fix: return jBoolean instead bool in jni

Doominika committed:
docs: change the order of exceptions

Doominika committed:
feat: add default value for version argument in setEntry method

Doominika committed:
fix: add ExceptionCheck in listKvdbs jni method

Doominika committed:
docs: add descriptions to EventType kvdb methods

Doominika committed:
fix: change HashMap.put method return value in jni

djenczewski commented on privmx-endpoint/src/main/cpp/modules/KvdbApi.cpp:
```diff
@@ -0,0 +1,627 @@
+                                env->FindClass("java/lang/NullPointerException"),
+                                "At least one of the keys has a null value."
+                        );
+                    }
```
this should break execution by return nullptr

djenczewski commented on privmx-endpoint/src/main/java/com/simplito/java/privmx_endpoint/modules/kvdb/KvdbApi.java:
```diff
@@ -0,0 +1,592 @@
+     * @returns 'true' if the KVDB has an entry with given key, 'false' otherwise
+     */
+    public native Boolean hasEntry(
+            String kvdbId,
```
change this type to primitive boolean

djenczewski commented on privmx-endpoint/src/main/cpp/modules/KvdbApi.cpp:
```diff
@@ -0,0 +1,658 @@
+                );
+
+                return ctx.bool2jBoolean(check_c && JNI_TRUE);
+            }
```
change this type to primitive jboolean

djenczewski commented on privmx-endpoint/src/main/cpp/model_native_initializers.cpp:
```diff
@@ -786,5 +786,151 @@ namespace privmx {
+            );
+        }
+
+        jobject serverKvdbEntryInfo2Java(JniContextUtils &ctx,
```
reformat this parameter list

Doominika committed:
fix: break execution after throwing exception in deleteEntries jni method

Doominika committed:
refactor: change return type of hasEntry method from Boolean to boolean

Doominika committed:
chore: reformat parameter lists in kvdb-related functions

filipg43 reviewed (commented):
"Kvdb" in all descriptions should be spelled "KVDB" since it's an abbreviation.

Doominika committed:
docs: change skip parameter description for kvdb methods

Doominika committed:
docs: use proper name for KVDB

Doominika committed:
docs: use proper name for KVDB

Doominika committed:
docs: add license to KvdbApi

Doominika committed:
docs(privmx-endpoint-extra): use proper name for KVDB

djenczewski committed:
sync with dev

# Conflicts:
#	privmx-endpoint-extra/src/main/java/com/simplito/java/privmx_endpoint_extra/events/EventType.java
#	privmx-endpoint-extra/src/main/java/com/simplito/java/privmx_endpoint_extra/lib/BasicPrivmxEndpoint.java
#	privmx-endpoint-extra/src/main/java/com/simplito/java/privmx_endpoint_extra/lib/PrivmxEndpoint.java
#	privmx-endpoint-extra/src/main/java/com/simplito/java/privmx_endpoint_extra/model/Modules.java
#	privmx-endpoint/src/main/cpp/CMakeLists.txt
#	privmx-endpoint/src/main/cpp/model_native_initializers.cpp
#	privmx-endpoint/src/main/cpp/model_native_initializers.h
